core/src/main/java/com/datastax/oss/driver/internal/core/metrics/DropwizardMetricUpdater.java

```diff
@@ -15,62 +15,69 @@
  */
 package com.datastax.oss.driver.internal.core.metrics;
 
+import com.codahale.metrics.Metric;
 import com.codahale.metrics.MetricRegistry;
 import com.codahale.metrics.Timer;
 import com.datastax.oss.driver.api.core.config.DefaultDriverOption;
 import com.datastax.oss.driver.api.core.config.DriverConfigProfile;
+import com.datastax.oss.driver.api.core.type.reflect.GenericType;
 import java.time.Duration;
 import java.util.Set;
 import java.util.concurrent.TimeUnit;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public abstract class MetricUpdaterBase<MetricT> implements MetricUpdater<MetricT> {
+public abstract class DropwizardMetricUpdater<MetricT> implements MetricUpdater<MetricT> {
 
-  private static final Logger LOG = LoggerFactory.getLogger(MetricUpdaterBase.class);
+  private static final Logger LOG = LoggerFactory.getLogger(DropwizardMetricUpdater.class);
 
   protected final Set<MetricT> enabledMetrics;
-  protected final MetricRegistry metricRegistry;
+  protected final MetricRegistry registry;
 
-  protected MetricUpdaterBase(Set<MetricT> enabledMetrics, MetricRegistry metricRegistry) {
+  protected DropwizardMetricUpdater(Set<MetricT> enabledMetrics, MetricRegistry registry) {
     this.enabledMetrics = enabledMetrics;
-    this.metricRegistry = metricRegistry;
+    this.registry = registry;
   }
 
   protected abstract String buildFullName(MetricT metric);
 
   @Override
   public void incrementCounter(MetricT metric, long amount) {
     if (enabledMetrics.contains(metric)) {
-      metricRegistry.counter(buildFullName(metric)).inc(amount);
+      registry.counter(buildFullName(metric)).inc(amount);
     }
   }
 
   @Override
   public void updateHistogram(MetricT metric, long value) {
     if (enabledMetrics.contains(metric)) {
-      metricRegistry.histogram(buildFullName(metric)).update(value);
+      registry.histogram(buildFullName(metric)).update(value);
     }
   }
 
   @Override
   public void markMeter(MetricT metric, long amount) {
     if (enabledMetrics.contains(metric)) {
-      metricRegistry.meter(buildFullName(metric)).mark(amount);
+      registry.meter(buildFullName(metric)).mark(amount);
     }
   }
 
   @Override
   public void updateTimer(MetricT metric, long duration, TimeUnit unit) {
     if (enabledMetrics.contains(metric)) {
-      metricRegistry.timer(buildFullName(metric)).update(duration, unit);
+      registry.timer(buildFullName(metric)).update(duration, unit);
     }
   }
 
+  @SuppressWarnings("unchecked")
+  public <T extends Metric> T getMetric(MetricT metric, GenericType<T> expectedType) {
```

Comment:
`expectedType` is unused, shouldn't we at least check that the returned metric is an instance of it?

Reply:
Guava's type token doesn't have any method to do that. Understandably so, because it could only compare to the erased type of the object.
Which leads me to believe that `expectedType` is redundant here; I put it thinking of the `Class<T>`-based type token pattern, but really it doesn't help at all in this case, since we're going to suppress a warning anyway.